flexget/plugins/input/from_telegram.py

```diff
@@ -0,0 +1,226 @@
+            # This is the ID
+            update_id = message['update_id']
+
+            # Update the last ID for the Bot
+            logger.debug("Last Update set to {}", update_id)
+            task.simple_persistence[token + '_update_id'] = update_id
+
+            # We Don't care if it's not a message or no text
+            if (
+                'message' not in message
+                or 'text' not in message['message']
+                or 'chat' not in message['message']
+                or 'type' not in message['message']['chat']
+            ):
+                logger.debug("Invalid message discarted: {}", message)
+                continue
+
+            logger.debug("Income message: {}", message)
+
+            # Check Types
+            if types and message['message']['chat']['type'] not in types:
+                logger.debug("Ignoring message because of invalid type {}", message)
+                continue
+
+            # Create Base Entry
+            text = message['message']['text']
+            entry = Entry()
+            entry['title'] = text
+
+            # We need a url, so we add a dummy
+            entry['url'] = "http://localhost?update_id={}".format(str(update_id))
+
+            # Store the message if we need to use it in other plugins
+            entry['telegram_message'] = message['message']
+
+            # Check From
+            message_from = message['message']['from']
+            message_chat = message['message']['chat']
+
+            if whitelist:
+                for check in whitelist:
+                    if 'username' in check and check['username'] == message_from['username']:
+                        logger.debug("WhiteListing: Username {}", message_from['username'])
+                        break
+                    elif (
+                        'fullname' in check
+                        and check['fullname']['first'] == message_from['first_name']
+                        and check['fullname']['sur'] == message_from['last_name']
+                    ):
+                        logger.debug(
+                            "WhiteListing: Full Name {} {}",
+                            message_from['first_name'],
+                            message_from['last_name'],
+                        )
+                        break
+                    elif 'group' in check:
+                        if (
+                            message_chat['type'] == 'group'
+                            and message_chat['title'] == check['group']
+                        ):
+                            logger.debug("WhiteListing: Group {}", message_chat['title'])
+                            break
+                else:
+                    logger.debug("Ignoring message because of no whitelist match {}", message)
+                    continue
+
+            # Process the entry config
+            accept = True
+            if entry_config:
+                for field, regexp in entry_config.items():
```

Comment:
im still not sure why we need a regex engine internally when there's the regexp plugin. not sure we should mix the filtering with the input

Reply:
The idea is to filter the messages and build the input in the plugin, similar to the text plugin (inspired by). The message can be anything and this regex will help arranje the fields and also validate the message. With this we can use the plugin directly in discover, for example. Without this we would need thing like from_task to clean up the messages. IMO I don't think this plugin will be as useful as it is, without the entry. Like this the user can decide if he what's to use it, or not and use other method of parsing.